src/analysis/convex/simplicial_complex/basic.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import analysis.convex.topology
+
+/-!
+# Simplicial complexes
+
+In this file, we define simplicial complexes in `𝕜`-modules. A simplicial complex is a collection
+of simplices closed by inclusion (of vertices) and intersection (of underlying sets).
+
+We model them by a downward-closed set of affine independent finite sets whose convex hulls "glue
+nicely", each finite set and its convex hull corresponding respectively to the vertices and the
+underlying set of a simplex.
+
+## Main declarations
+
+* `simplicial_complex 𝕜 E`: A simplicial complex in the `𝕜`-module `E`.
+* `simplicial_complex.vertices`: The zero dimensional faces of a simplicial complex.
+* `simplicial_complex.facets`: The maximal faces of a simplicial complex.
+
+## TODO
+
+Simplicial complexes can be generalized to affine spaces once `convex_hull` has been ported.
+-/
+
+open finset set
+
+variables {𝕜 E ι : Type*} [ordered_ring 𝕜] [add_comm_group E] [module 𝕜 E]
```

Comment:
How about `ordered_field 𝕜`? And then adjust "module" to "vector space" in comments?
I think the definition here isn't sensible for e.g. `int` (the `convex_hull` condition will become trivial). For example even the simplicial complex containing the two 1-simplices `(0, 0) -- (2, 2)` and `(2, 0) -- (0, 2)` would qualify.

Reply:
Yury mentioned convexity of measures, which corresponds to `ℝ≥0`-convexity. So my goal is to generalize to `ordered_semiring` (but we need the affinity refactor to happen first). What behavior are you expecting from `ℤ`-convexity?

Reply:
I'm not expecting to give any meaning at all to a simplicial complex in a module over a ring, that's why I suggested making `𝕜` a field.